example/pauli/dialect.py

```diff
@@ -0,0 +1,3 @@
+from kirin import ir
+
+dialect_ = ir.Dialect("pauli")
```

Comment:
we usually do the reverse, name the file `_dialect.py` (because you are not interested in using this module outside the package)

name the dialect `dialect` so that you can directly use the module object as if it is the dialect (the APIs in Kirin is smart enough to pick it up)